File: app/src/main/java/com/sns/todo_app_android_fireabse/models/Todo.java
package com.sns.todo_app_android_fireabse.models;

import com.google.firebase.Timestamp;

public class Todo {

    private String id;
    private String name;
    private Timestamp timestamp;

    public Todo() {
    }

    public Todo(String id, String name, Timestamp timestamp) {
        this.id = id;
        this.name = name;
        this.timestamp = timestamp;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Timestamp getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(Timestamp timestamp) {
        this.timestamp = timestamp;
    }
}

File: app/src/main/java/com/sns/todo_app_android_fireabse/models/Todo.kt
package com.sns.todo_app_android_fireabse.models

import com.google.firebase.Timestamp
import java.io.Serializable

class Todo : Serializable {
    //     class Timestamp implements Serializable {}
    var id: String? = null
    var name: String? = null
    private var timestamp: Timestamp? = null

    // 空のコンストラクタを用意しないと不具合になる
//    constructor() {}
//    constructor(id: String?, name: String?, timestamp: Timestamp?) {
//        this.id = id
//        this.name = name
//        this.timestamp = timestamp
//    }
}